html/modules/custom/reliefweb_import/src/Service/WorkDayJobImporter.php

```diff
@@ -0,0 +1,326 @@
+        'client_id' => $client_id,
+        'client_secret' => $client_secret,
+        'refresh_token' => $refresh_token,
+      ],
+    ]);
+
+    if ($response->getStatusCode() !== 200) {
+      throw new \Exception('Failure with response code: ' . $response->getStatusCode());
+    }
+
+    try {
+      $token = json_decode($response->getBody()->getContents(), TRUE, flags: \JSON_THROW_ON_ERROR)['access_token'];
+    }
+    catch (\Exception $e) {
+      throw new \Exception('Unable to decode the auth token: ' . $e->getMessage());
+    }
+
+    return $token;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function importJobs(int $limit = 50): void {
+    $jobs = $this->getDocuments($limit);
+
+    // Switch to proper user and import XML.
+    $uid = $this->settings['uid'] ?? 2;
+
+    /** @var \Drupal\user\UserInterface $account */
+    $account = $this->entityTypeManager->getStorage('user')->load($uid);
+    $account->addRole('job_importer');
+    $this->accountSwitcher->switchTo($account);
+
+    $this->importDocuments($jobs);
+
+    // Restore user account.
+    $this->accountSwitcher->switchBack();
+  }
+
+  /**
+   * Retrieve documents from the WorkDay.
+   *
+   * @param int $limit
+   *   Number of documents to fetch.
+   *
+   * @return array
+   *   List of documents keyed by IDs.
+   */
+  public function getDocuments(int $limit = 50): array {
+    $this->getLogger()->info('Retrieving documents from the WorkDay.');
+
+    $documents = [];
+
+    try {
+      $timeout = $this->settings['timeout'] ?? 10;
+      $base_url = $this->settings['base_url'] ?? '';
+      $tenant = $this->settings['tenant'] ?? '';
+      $url = $base_url . '/ccx/api/recruiting/v4/' . $tenant . '/jobPostings';
+
+      $auth = $this->getAuthToken();
+
+      $response = $this->httpClient->request('GET', $url, [
+        'connect_timeout' => $timeout,
+        'timeout' => $timeout,
+        'headers' => [
+          'Content-Type' => 'application/json',
+          'Accept' => 'application/json',
+          'Authorization' => 'Bearer ' . $auth,
+        ],
+      ]);
+
+      if ($response->getStatusCode() !== 200) {
+        throw new \Exception('Failure with response code: ' . $response->getStatusCode());
+      }
+
+      $jobs = json_decode($response->getBody()->getContents(), TRUE);
+      foreach ($jobs['data'] as $job) {
+        // Skip jobs without title or URL.
+        if (empty($job['title']) || empty($job['url'])) {
+          continue;
+        }
+
+        $documents[] = [
+          'id' => $job['id'],
+          'field_job_closing_date' => $job['startDate'] ?? '',
+          'job_type' => $job['jobType']['descriptor'] ?? '',
+          'company' => $job['company']['descriptor'] ?? '',
+          'time_type' => $job['timeType']['descriptor'] ?? '',
+          'field_city' => $job['primaryLocation']['descriptor'] ?? '',
+          'field_country' => [$job['primaryLocation']['country']['descriptor'] ?? ''],
```

Comment:
This can result in `field_country => ['']` which is probably not the expected value.